ledger/src/blockstore.rs

```diff
@@ -2971,20 +2971,19 @@ impl Blockstore {
     }
 
     // Returns slots connecting to any element of the list `slots`.
-    pub fn get_slots_since(&self, slots: &[u64]) -> Result<HashMap<u64, Vec<u64>>> {
-        // Return error if there was a database error during lookup of any of the
-        // slot indexes
-        let slot_metas: Result<Vec<Option<SlotMeta>>> =
-            slots.iter().map(|slot| self.meta(*slot)).collect();
-
-        let slot_metas = slot_metas?;
-        let result: HashMap<u64, Vec<u64>> = slots
+    pub fn get_slots_since(&self, slots: &[u64]) -> Result<Vec<(u64, Vec<u64>)>> {
```

Comment:
So this means we will have a public API change, can I know how we usually handle public API change?  Or the owners of validators will know the new release breaks their code and they just change accordingly?

Reply:
So this function is technically declared as `pub`; however, we don't strict API guidelines here similar to what we have with say the RPC methods.

Most validator operators use the software as-is, and as such, this signature change isn't even on their radar. For those that might be tweaking things, they are operating with the understanding that the codebase is still under heavy churn